src/builders/typescript/typescript-direct-builder.ts

```diff
@@ -0,0 +1,186 @@
+                    return;
+                }
+                if (allowedExtensions == null) {
+                    allowedExtensions = [".ts", ".tsx"];
+                }
+                let typescriptFiles = globbedFiles.filter(file => {
+                    let extension = path.extname(file);
+                    return allowedExtensions == null || allowedExtensions.indexOf(extension) !== -1;
+                });
+                resolve(typescriptFiles);
+            });
+        });
+    }
+
+    protected LoadTsConfig(production: boolean): TsConfig {
+        let tsConfigName = production ?
+            Configuration.GulpConfig.TypeScriptConfig.Production :
+            Configuration.GulpConfig.TypeScriptConfig.Development;
+
+        // Get tsconfig.json path
+        let tsConfigPath = path.normalize(`${process.cwd()}/${tsConfigName}`);
+
+        // Read its contents
+        return require(tsConfigPath);
+    }
+
+    public async Lint(files: string[], lintDefinitions: boolean = false): Promise<LintResult[]> {
+        return new Promise<LintResult[]>((resolve, reject) => {
+            let tslintConfigPath = path.normalize(`${process.cwd()}/tslint.json`);
+            try {
+                let configurationFile = Linter.loadConfigurationFromPath(tslintConfigPath);
+                let options: ILinterOptionsRaw = {
+                    formatter: "verbose",
+                    configuration: configurationFile
+                };
+                let lintResults: LintResult[] = [];
+                for (let file of files) {
+                    let stats = fs.lstatSync(file);
+                    if (stats.isFile()) {
+                        let contents = fs.readFileSync(file, "utf8");
+                        let linter = new Linter(file, contents, options, this.typescriptProgram);
+                        let result = linter.lint();
+                        lintResults.push(result);
+                    }
+                }
+                resolve(lintResults);
+            }
+            catch (error) {
+                console.error(error);
+                reject(error);
+            }
+        });
+    }
+
+    public async LintAll(production: boolean, lintDefinitions: boolean = false): Promise<LintResult[]> {
+        let tsConfigFromJson: TsConfig = this.LoadTsConfig(production);
+        let globbedFiles = await this.GlobTypescriptFiles(tsConfigFromJson.include, tsConfigFromJson.exclude);
+
+        if (!lintDefinitions) {
+            let dts = ".d.ts";
+            let dtsLength = dts.length;
+            globbedFiles = globbedFiles.filter(file => {
+                return file.substr(file.length - dtsLength) !== dts;
+            });
+        }
+
+        return await this.Lint(globbedFiles);
+    }
+
+    public PrintDiagnostics(diagnostics: ts.Diagnostic[], logger: Logger) {
+        for (let diagnostic of diagnostics) {
+            if (diagnostic.file == null || diagnostic.file.isDeclarationFile) {
```

Comment:
https://github.com/QuatroCode/simplr-gulp/pull/40/commits/502effe881e642a13e1ef3e389e64f360eca79c3 Done.
Now `tsConfig.json` value for `skipDefaultLibCheck` is respected and used while printing diagnostics and definition files are not filtered out while linting, to have all of them if needed.